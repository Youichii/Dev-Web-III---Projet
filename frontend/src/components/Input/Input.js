import PropTypes from 'prop-types';

const Input = ( { type, name, min, max, placeholder, setFunc, pattern, title, step} ) => {
    return (<input className="input_modif" step={step} pattern={pattern} required title={title} type={type} name={name} minLength={min} maxLength={max} placeholder={placeholder} onChange={(e) => {
                                setFunc(e.target.value)
                            }}/>
    )
}

Input.defaultProps = {
    type : 'text',
    min : 4,
    max : 10,
    placeholder : 'texte'
}

export default Input