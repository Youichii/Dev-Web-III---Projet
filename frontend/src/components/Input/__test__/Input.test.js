import React from 'react';
import ReactDOM from 'react-dom';
import Input from '../Input';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

//npm test to start all tests
afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Input />, div)
})