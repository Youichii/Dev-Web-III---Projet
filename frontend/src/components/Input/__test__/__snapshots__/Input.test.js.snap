// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates snapeshot with default props 1`] = `
<input
  data-testid="input"
  maxLength={10}
  minLength={4}
  onChange={[Function]}
  placeholder="texte"
  required={true}
  type="text"
/>
`;
